Frontend/ui-customer/src/Flow.purs

```diff
@@ -346,60 +346,93 @@ handleDeepLinks mBGlobalPayload skipDefaultCase = do
+                if isJust queryParam.option then do
+                  let
+                    videoList = RC.safetyBannerVideoConfigData (DS.toLower $ getValueToLocalStore CUSTOMER_LOCATION) $ fetchLanguage $ getLanguageLocale languageKey
+                  modifyScreenState $ NammaSafetyScreenStateType (\safetyScreen -> safetyScreen { data { videoList = videoList }, props { showVideoView = true, educationViewIndex = Just 0, fromBannerLink = true } })
+                else
+                  modifyScreenState $ NammaSafetyScreenStateType (\safetyScreen -> safetyScreen { props { fromDeepLink = true } })
+              _ -> pure unit
+            hideSplashAndCallFlow safetyEducationFlow
+          "mt" -> hideSplashAndCallFlow metroTicketBookingFlow
+          "shareRide" ->  do
+            case globalPayload ^. _payload ^. _deepLinkJSON of
+              Just (CTA.QueryParam queryParam) -> do
+                case queryParam.rideId of
+                  Just rId -> do
+                    res <- lift $ lift $ HelpersAPI.callApi $ GetManuallySharedDetailsReq rId
+                    case res of
+                      Right (GetManuallySharedDetailsRes rideDetails) -> do
+                        let follower = { name : Just rideDetails.customerName, bookingId : rideDetails.bookingId, mobileNumber : rideDetails.customerPhone, priority : 0 }
+                        modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen { data { followers = Just [follower], manuallySharedFollowers = Just [follower] } })
+                        updateFollower false true Nothing
+                      Left _ -> currentFlowStatus
+                  _ -> currentFlowStatus
+              _ -> pure unit
+          _ -> do
+            case breakPrefixAndId screen of
+              Just ( Tuple "metroBooking" bookingId )-> do
+                case bookingId of
+                  Just id -> hideSplashAndCallFlow $ viewTicketDetialsFlow (Just id)
+                  _ -> pure unit
+              _ -> if skipDefaultCase then pure unit else currentFlowStatus
+        Nothing -> currentFlowStatus
     Nothing -> do
       let
         mBPayload = getGlobalPayload Constants.globalPayload
       case mBPayload of
         Just _ -> handleDeepLinks mBPayload skipDefaultCase
         Nothing -> pure unit
 
+
+handleWidgetData :: String -> FlowBT String Unit
+handleWidgetData urlData = 
+  case urlData of 
+    _ | urlData == "Home" || urlData == "Work" -> do
+      (SavedLocationsListRes savedLocationResp )<- FlowCache.updateAndFetchSavedLocations true 
+      let items =  spy  "getSavedLocations" $ AddNewAddress.getSavedLocations savedLocationResp.list
+          maybeItem = find (\x -> x.tag == urlData  ) items
+      case maybeItem of 
+        Just item -> do
+          updateLocalStage GoToConfirmLocation
+          let 
+              lastKnowLat =  (fromMaybe 0.0 $ fromString $ getValueToLocalNativeStore LAST_KNOWN_LAT) 
```

Comment:
We should not have default lat longs as 0.0. In a case where lst lons are nothing, Better to redirect the user to the homescreen rather than pointing them to 0.0.